core/backend/src/Entity.ts

```diff
@@ -100,6 +100,61 @@ export class Entity {
    * @internal
    */
   public static get isGeneratedClass() { return false; }
+
+  // NOTE: will also consider using generated collectReferences everywhere and a separate collectJsonPropertyReferenceIds
+  /** Get the Ids of this element's *references*. A *reference* is an element that this element references.
+   * This is important for cloning operations but can be useful in other situations as well.
+   * @note In the next breaking change, the behavior of this function will change to return a EntityReferenceSet
+   *       which does not iterate plain Id64s, see [EntityReferenceSet]($backend).
+   * @see collectReferenceIds
+   * @beta
+   */
+  public getReferenceIds(): Set<Id64String> {
+    const referenceIds = new Set<Id64String>();
+    this.collectReferenceIds(referenceIds); // eslint-disable-line deprecation/deprecation
+    return referenceIds;
+  }
+
+  /** This is the intended and future behavior of [[getReferenceIds]].
+   * In the next breaking change it will replace getReferenceIds, and is already used by the transformer
+   * @internal
+   */
+  public getReferenceConcreteIds(): EntityReferenceSet {
+    const referenceIds = new EntityReferenceSet();
+    this.collectReferenceConcreteIds(referenceIds);
+    return referenceIds;
+  }
+
+  // FIXME: test (in ClassRegistry?) that a class with its own implementation of this still gets the expected concrete and non-concrete ids
```

Comment:
Did you intend to leave this FIXME in the code?

Reply:
that is done (see ClassRegistry.test.ts changes), will remove this comment

Reply:
done